lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -0,0 +1,74 @@
+//===- Expressions.cpp - Slang expression conversion ----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "ImportVerilogInternals.h"
+#include "slang/syntax/AllSyntax.h"
+
+using namespace circt;
+using namespace ImportVerilog;
+
+// NOLINTBEGIN(misc-no-recursion)
```

Comment:
Why is it fine to always do a recursive call to traverse expressions? Isn't this like using recursion to traverse the def-use chain?

Reply:
Good point. I'm not sure it actually is fine. Well, probably in the beginning, but this probably has to be come that worklist-based approach you were suggesting. Not sure how exactly that would work -- if you'd push all child expressions of a node onto the worklist and then check some hash map to get the actual child expressions.

If I'm not mistaken, Slang uses a recursive descent parser to construct the AST. That would mean any expression tree being visited here has been visited recursively while parsing already. That might indicate that parsing is going to fail first or at the same time as this lowering code.

I'm going to create an issue to turn this into a worklist once the remaining Slang integration PRs have landed.